src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,63 @@
+package duke;
+
+import duke.commands.*;
+import duke.exceptions.NoSuchCommandException;
+import duke.tasks.DeadlinesTask;
+import duke.tasks.EventsTask;
+import duke.tasks.Task;
+import duke.tasks.ToDosTask;
+
+
+public class Parser {
+    Parser(){
+
+    }
+
+    public static Command parse(String input) throws NoSuchCommandException {
+        switch(input.split(" ")[0]) {
```

Comment:
```suggestions
switch(...) {
case "list":
    ...
...
}
```
Be careful about this, for this module the requirement for writing switch cases is little different from what we encountered before.